components/Post/Post.tsx

```diff
@@ -0,0 +1,96 @@
+'use client';
+
+import { useEffect, useState } from 'react';
+import { FiHeart as HeartIcon } from 'react-icons/fi';
+import { GoPaperAirplane as ShareIcon } from 'react-icons/go';
+import { LuUserCircle as ProfileIcon } from 'react-icons/lu';
+import { UUID } from 'crypto';
+import supabase from '@/supabase/client';
+import styles from './postStyles.module.css';
+
+interface PostI {
+  uuid: UUID;
+  created_at: string;
+  description: string;
+  username: string;
+  image_url: string;
+  like_count: number;
+}
+
+export default function Post() {
+  // Define a state variable to store the fetched data
+  const [postData, setPostData] = useState<PostI[] | undefined>(undefined);
+
+  async function fetchPostData() {
+    // TODO: Fetch data using the supabase client
+    const { data, error } = await supabase.from('posts').select();
+    if (error) {
+      console.error(error);
+      return;
+    }
+    // TODO: Format data into the Post type
+    console.log(data);
+    const format: PostI[] = data?.map(
+      val =>
+        ({
+          uuid: val.uuid as UUID,
+          created_at: val.created_at as string,
+          description: val.description as string,
+          username: val.username as string,
+          image_url: val.image_url as string,
+          like_count: val.like_count as number,
+        })!,
+    );
+    return format;
```

Comment:
Given that we only want to display one post on screen at once, it may make more sense to store a single Post item state, rather than a full array of posts.